### PR TITLE
Added signatures, tests, and docs for append_array function (2) (Issue 481)

### PR DESCRIPTION
**Pull request by bbbales2**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py src/test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary

Replaces https://github.com/stan-dev/stan/pull/2343 (cause I don't have stan-dev access now). It is also rebased on the latest Stan develop branch.

Added append_array function hooks into Stan (stan-dev/math#481) 

#### How to Verify

./runTests.py src/test/unit/lang/parser/matrix_functions_test.cpp

#### Documentation

Added under "Array Concatenation" in src/docs/stan-reference/functions.tex

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): University of California, Santa Barbara

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by bbbales2** [comment]
@seantalts For the errors on Travis, should I just merge the latest develop in here?

**Comment by bob-carpenter** [comment]
The test framework will merge with latest develop, so you might as well pull the latest develop into your branch locally to stay up to date.

**Comment by seantalts** [comment]
Yeah, I think that will help as well - develop has changes to the travis config that slow the tests down but prevent travis from running out of memory

**Comment by bbbales2** [comment]
Checks passed, this should be good to review.

File: src/docs/stan-reference/functions.tex [hunk 1]
```diff
@@ -1818,6 +1817,29 @@ \section{Array Broadcasting}\label{array-broadcasting.section}
 \code{k} in \code{1:4}, \code{m} in \code{1:5}, and \code{n} in
 \code{1:6}.
 
+%
+\section{Array Concatenation}\label{array-concatenation.section}
+
+\begin{description}
+\fitem{T}{append\_array}{T \farg{x}, T \farg{y}}{
+  Return the concatenation of two arrays in the order of the arguments.
+  T must be an N-dimensional array of any Stan type (with a maximum N of 8). All
```
Comment (bob-carpenter) [inline src/docs/stan-reference/functions.tex]: I think this is actually a maximum N of 7, as the loop goes from 1 through `N < 8`.

**Comment by bob-carpenter** [comment]
Oops, can't fix myself as this isn't on our repo.  Darn.  I'm not liking this remote repo dev stuff.

**Comment by bob-carpenter** [comment]
I'm going to merge and then leave a note in the next doc issue to fix this to help clear the jam of pull requests.

**Pull request merged**